packages/account/src/utils/consolidate-coins.ts

```diff
@@ -0,0 +1,241 @@
+  const baseAssetId = await account.provider.getBaseAssetId();
+  const isBaseAsset = assetId === baseAssetId;
+
+  const batchSize = maxInputs;
+  const numberOfFundingCoins = maxInputs;
+
+  let funding: Coin[] = [];
+  let dust: Coin[] = [];
+
+  // We get the largest coin/s for funding purposes
+  if (isBaseAsset) {
+    const coins = await getAllCoins(account, baseAssetId).then(sortCoins);
+    funding = coins.slice(0, numberOfFundingCoins);
+    dust = coins.slice(numberOfFundingCoins);
+  } else {
+    funding = await getAllCoins(account, baseAssetId)
+      .then(sortCoins)
+      .then((coins) => coins.slice(0, numberOfFundingCoins));
+    dust = await getAllCoins(account, assetId).then(({ coins }) => coins);
+  }
+
+  // There a better way of detecting whether the account has enough funds to consolidate
+  if (funding.length === 0) {
+    throw new FuelError(
+      ErrorCode.INSUFFICIENT_FUNDS,
+      `Insufficient funds to consolidate.\n\tAsset ID: ${baseAssetId}\n\tOwner: ${account.address.toB256()}`
+    );
+  }
+
+  const batches = [
+    ...splitEvery(batchSize, funding),
+    // We leave one coin for the funding coin
+    ...splitEvery(batchSize - 1, dust),
+  ];
+
+  const txs: ScriptTransactionRequest[] = batches.map((batch) => {
+    const request = new ScriptTransactionRequest({
+      scriptData: '0x',
+    });
+
+    // Add our dust coins as inputs
+    request.addResources(batch);
+
+    return request;
+  });
+
+  const submitAll: SubmitAllCallback = async (opts = {}) => {
+    const txResponses: TransactionResult<TransactionType.Script>[] = [];
+    let previousTx: { transactionId: string; outputs: Output[] } | undefined;
+
+    for (let i = 0; i < txs.length; i++) {
+      let currentTx = txs[i];
+      const step = i + 1;
+
+      if (previousTx) {
+        const coin = createOuputCoin({
+          transactionId: previousTx.transactionId,
+          outputs: previousTx.outputs,
+          baseAssetId,
+        });
+
+        // Add the funding coin to the current tx
+        currentTx.addResource(coin);
+      }
+
+      // Populate the tx with predicate data
+      if (
+        'populateTransactionPredicateData' in account &&
+        typeof account.populateTransactionPredicateData === 'function'
+      ) {
+        currentTx = account.populateTransactionPredicateData(currentTx);
+        currentTx = await account.provider.estimatePredicates(currentTx);
+      }
+
+      // Calculate gas and fee
+      const fee = calculateGasFee({
+        gasPrice,
+        gas: currentTx.calculateMinGas(chainInfo),
+        priceFactor: chainInfo.consensusParameters.feeParameters.gasPriceFactor,
+        tip: currentTx.tip,
+      });
+
+      currentTx.maxFee = fee;
+      currentTx.gasLimit = bn(1000);
+
+      opts.onTransactionStart?.({
+        tx: currentTx,
+        step,
+        assetId,
+        transactionId: currentTx.getTransactionId(chainId),
+      });
```

Comment:
was this `onTransactionStart` needed for connectors integration? if yes for what?